docs/src/systems/implicit_incompressible_sph.md

```diff
@@ -0,0 +1,293 @@
+
+Therefore, the following equation needs to be fulfilled:
+```math
+\Delta t ^2 \sum_j m_j \left(  \frac{\bm{F}_i^p(t)}{m_i} - \frac{\bm{F}_j^p(t)}{m_j} \right) \nabla W_{ij}(t) = \rho_0 - \rho_i^{\text{adv}}.
+```
+
+This expression is derived by substituting the reference density $\rho_0$ for
+$\rho_i(t+\Delta t)$ in the discretized continuity equation and inserting the definitions of
+the predicted velocity $\bm{v}_{ij}(t+\Delta t)$ and predicted density $\rho_i^{\text{adv}}$.
+
+The pressure force is defined as:
+
+```math
+\bm{F}_i^p(t) = m_i/rho_i  \nabla p_i = m_i \sum_j m_j \left( \frac{p_i(t)}{\rho_i^2(t)} - \frac{p_j(t)}{\rho_j^2(t)} \right) \nabla W_{ij}
+```
+
+Substituting this definition into the equation, yields a linear system $\bm{A}(t) \bm{p}(t) = \bm{b}(t)$
+with one equation and one unknown pressure value for each particle.
+For ``n`` particles, this is a linear system with ``n`` equations and ``n`` unknowns:
+
+```math
+\sum_j a_{ij} p_j = b_i = \rho_0 - \rho_i^{\text{adv}}.
+```
+
+To solve for the pressure values, a relaxed Jacobi scheme is used.
+This is an iterative numerical method to solve a linear system $Ap=b$. In each iteration, the
+new values of $p$ are obtained as computed as
+
+```math
+p_i^{(k+1)} = (1-\omega) p_i^{(k)} + \omega \left( \frac{1}{a_{ii}} \left( b_i - \sum_{j \neq i} a_{ij} p_j^{(k)} \right)\right)
+```
+
+Substituting the right-hand side, we obtain
+
+```math
+p_i^{l+1} = (1-\omega) p_i^l + \omega \frac{\rho_0 - \rho_i^{\text{adv}} - \sum_{j \neq i} a_{ij}p_j^l}{a_{ii}}.
+```
+
+Therefore the diagonal elements $a_{ii}$ and the sum $\sum_{j \neq i} a_{ij}p_j^l$ need to
+be determined.
+This can be done efficiently by separating the pressure force acceleration into two
+components: One that describes the influence of particle i's own pressure on its
+displacement, and one that describes the displacement due to the pressure values of the
+neighboring particles $j$.
+
+The pressure acceleration is given by:
+```math
+\Delta t^2 \frac{\bm{F}_i^p}{m_i} = -\Delta t^2 \sum_j m_j \left( \frac{p_i}{\rho_i^2} + \frac{p_j}{\rho_j^2} \right)\nabla W_{ij} = \left( - \Delta t^2 \sum_j \frac{m_j}{\rho_i^2} \nabla W_{ij} \right) p_i + \sum_j - \Delta t^2 \frac{m_j}{\rho_j^2} \nabla W_{ij}p_j
+```
+
+The $d_{ii}p_i$ value describes the displacement of particle ``i`` because of the particle ``i``
+and $d_{ij}p_j$ describes the influence from the neighboring particles ``j``.
+Using this new values the linear system can be rewritten as
+
+```math
+\rho_0 - \rho_i^{\text{adv}} = \sum_j m_j \left( d_{ii}p_i + \sum_j d_{ij}p_j - d_{jj}p_j - \sum_k d_{jk}p_k \right) \nabla W_{ij}
+```
+
+where $k$ stands for the neighbor particles of the neighbor particle ``j`` from ``i``.
+So the sum over the neighboring pressure values $p_j$ also includes the pressure values $p_i$,
+since ``i`` is a neighbor of ``j``
+To separate this sum it can be written as
+
+```math
+\sum_k d_{jk} p_k = \sum_{k \neq i} d_{jk} p_k + d_{ji} p_i
```

Comment:
```suggestion
\sum_k d_{jk} p_k = \sum_{k \neq i} d_{jk} p_k + d_{ji} p_i.
```
Please check all other equations as well for punctuation. I will not comment on every equation.